src/linker/Linker.Steps/ValidateVirtualMethodAnnotationsStep.cs

```diff
@@ -41,8 +41,8 @@ protected override void Process ()
 		void ValidateMethodRequiresUnreferencedCodeAreSame (MethodDefinition method, MethodDefinition baseMethod)
 		{
 			var annotations = Context.Annotations;
-			bool methodHasAttribute = annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (method);
-			if (methodHasAttribute != annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (baseMethod)) {
+			bool methodHasAttribute = annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (method) || method.DeclaringType is not null && annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (method.DeclaringType);
+			if (methodHasAttribute != annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (baseMethod) || baseMethod.DeclaringType is not null && annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (baseMethod.DeclaringType)) {
```

Comment:
We already have a few places that perform similar checks to ask if a method is RUC (either directly or due to the type annotations). Elsewhere we still call TryGetEffectiveRequiresUnreferencedCodeAttributeOnType to deal with nested types but I think we are getting rid of that. Could we introduce a helper that encapsulates this logic to make sure it's consistent everywhere we call it?

Reply:
Replaced with DoesMethodRequireUnreferencedCode method